### ISSUE TITLE
[visualization] some warnings will be displayed with VTK 9

### ISSUE THREAD

**Issue by UnaNancyOwen** [PointCloudLibrary/pcl#4830]
**Describe the bug**

If you build PCL with VTK9, you will see some warnings when app runtime.
These warnings are caused by the use of deprecated features to be removed in the future of VTK.

**To Reproduce**

1. install [PCL 1.12.0 RC1](https://github.com/PointCloudLibrary/pcl/releases/tag/pcl-1.12.0-rc1) with VTK 9.
2. run [tutorial code](https://github.com/PointCloudLibrary/pcl/tree/master/doc/tutorials/content/sources/pcl_visualizer) of pcl visualizer.

**Screenshots/Code snippets**

* screen shot

  ![2021-07-02_09h50_03](https://user-images.githubusercontent.com/816705/124205673-ab091b00-db1c-11eb-8745-e8a955b93e72.png)

* console

```
2021-07-02 09:48:16.193 (   0.273s) [                ]vtkOpenGLPolyDataMapper:271   WARN| vtkOpenGLPolyDataMapper::GetVertexShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetVertexShaderCode instead.
2021-07-02 09:48:16.220 (   0.299s) [                ]vtkOpenGLPolyDataMapper:263   WARN| vtkOpenGLPolyDataMapper::SetVertexShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetVertexShaderCode instead.
2021-07-02 09:48:16.223 (   0.303s) [                ]vtkOpenGLPolyDataMapper:301   WARN| vtkOpenGLPolyDataMapper::GetGeometryShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetGeometryShaderCode instead.
2021-07-02 09:48:16.227 (   0.307s) [                ]vtkOpenGLPolyDataMapper:293   WARN| vtkOpenGLPolyDataMapper::SetGeometryShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetGeometryShaderCode instead.
2021-07-02 09:48:16.234 (   0.313s) [                ]vtkOpenGLPolyDataMapper:286   WARN| vtkOpenGLPolyDataMapper::GetFragmentShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetFragmentShaderCode instead.
2021-07-02 09:48:16.239 (   0.318s) [                ]vtkOpenGLPolyDataMapper:278   WARN| vtkOpenGLPolyDataMapper::SetFragmentShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetFragmentShaderCode instead.
2021-07-02 09:48:16.244 (   0.324s) [                ]vtkOpenGLPolyDataMapper:271   WARN| vtkOpenGLPolyDataMapper::GetVertexShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetVertexShaderCode instead.
2021-07-02 09:48:16.252 (   0.332s) [                ]vtkOpenGLPolyDataMapper:263   WARN| vtkOpenGLPolyDataMapper::SetVertexShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetVertexShaderCode instead.
2021-07-02 09:48:16.260 (   0.340s) [                ]vtkOpenGLPolyDataMapper:301   WARN| vtkOpenGLPolyDataMapper::GetGeometryShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetGeometryShaderCode instead.
2021-07-02 09:48:16.268 (   0.348s) [                ]vtkOpenGLPolyDataMapper:293   WARN| vtkOpenGLPolyDataMapper::SetGeometryShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetGeometryShaderCode instead.
2021-07-02 09:48:16.278 (   0.358s) [                ]vtkOpenGLPolyDataMapper:286   WARN| vtkOpenGLPolyDataMapper::GetFragmentShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetFragmentShaderCode instead.
2021-07-02 09:48:16.287 (   0.367s) [                ]vtkOpenGLPolyDataMapper:278   WARN| vtkOpenGLPolyDataMapper::SetFragmentShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetFragmentShaderCode instead.
```
 
* vtkOutputWindow

```
Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 271
vtkOpenGLPolyDataMapper::GetVertexShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetVertexShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 263
vtkOpenGLPolyDataMapper::SetVertexShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetVertexShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 301
vtkOpenGLPolyDataMapper::GetGeometryShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetGeometryShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 293
vtkOpenGLPolyDataMapper::SetGeometryShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetGeometryShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 286
vtkOpenGLPolyDataMapper::GetFragmentShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetFragmentShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 278
vtkOpenGLPolyDataMapper::SetFragmentShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetFragmentShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 271
vtkOpenGLPolyDataMapper::GetVertexShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetVertexShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 263
vtkOpenGLPolyDataMapper::SetVertexShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetVertexShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 301
vtkOpenGLPolyDataMapper::GetGeometryShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetGeometryShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 293
vtkOpenGLPolyDataMapper::SetGeometryShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetGeometryShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 286
vtkOpenGLPolyDataMapper::GetFragmentShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::GetFragmentShaderCode instead.

Generic Warning: In C:\VTK-9.0.2\Rendering\OpenGL2\vtkOpenGLPolyDataMapper.cxx, line 278
vtkOpenGLPolyDataMapper::SetFragmentShaderCode was deprecated for VTK 9.0 and will be removed in a future version.  Use vtkOpenGLShaderProperty::SetFragmentShaderCode instead.
```

**Your Environment (please complete the following information):**

 - OS: Widnows 10
 - Compiler: Visual Studio 2019
 - PCL Version: PCL 1.12.0 RC1
 - VTK Version: VTK 9.0.2

**Possible Solution**

You can temporarily work around this problem by inserting this code snippet into user code.

```cpp
pcl::visualization::PCLVisualizer::Ptr viewer = pcl::make_shared<pcl::visualization::PCLVisualizer>( "Point Cloud Viewer" );
viewer->getRenderWindow()->GlobalWarningDisplayOff(); // Add This Line
```

**Comment by kunaltyagi** [comment]
Pinging @larshg 

TLDR:
Need to replace `vtkOpenGLPolyDataMapper` with `vtkOpenGLShaderProperty` to use
*  `{Get,Set}FragmentShaderCode `
* `{Get,Set}VertexShaderCode `
* `{Get,Set}GeometryShaderCode `

**Comment by kunaltyagi** [comment]
Questions in my head:
* Can we move this to 1.12.1 if no API/ABI is affected?
* Should we just disable warnings?

**Comment by larshg** [comment]
I don't think its something that we use explicitly. I at least cannot find any vktOpenGLPolyDataMapper, only vtkPolyDataMapper.
I think its a warning from VTK internal use, please see https://discourse.vtk.org/t/vtk-9-0-rc1/2916.

I would say, that its up to the users to silence those warnings - or would you add it to the PCLVisualizer?

**Comment by kunaltyagi** [comment]
Oh, ok, we can ignore them. For the binaries, we should explicitly silence them, and for the libraries, add this to release notes that the warnings are not ours.

**Comment by UnaNancyOwen** [comment]
> You can ignore the warnings, they are due to backwards compatibility code where VTK calls deprecated code itself. Turning on VTK_LEGACY_REMOVE in CMake I think gets rid of them.
> https://discourse.vtk.org/t/vtk-9-0-rc1/2916/3

If there are no problem and no bad effect turning VTK_LEGACY_REMOVE to ON in PCL, we can turn it on in VTK included in PCL All-in-one Installer.

**Comment by kunaltyagi** [comment]
I don't know about switching on the flag universally because that also hides the warnings in user-only code as well

**Comment by UnaNancyOwen** [comment]
This is report. I tried this https://github.com/PointCloudLibrary/pcl/issues/4830#issuecomment-873326759.
I built PCL that linked VTK with VTK_LEGACY_REMOVE turned ON. But, the user code that uses it did not work well. Therefore, this idea does not seem good.

**Comment by kunaltyagi** [comment]
Did you try VTK_LEGACY_REMOVE with pcl_visualizer?

**Comment by UnaNancyOwen** [comment]
> Did you try VTK_LEGACY_REMOVE with pcl_visualizer?

Yes, This is the report of that case. https://github.com/PointCloudLibrary/pcl/issues/4830#issuecomment-873332593

**Comment by kunaltyagi** [comment]
I don't think any action is needed from our end. Let's close the issue?
